components/expenses/PayoutBankInformationForm.js

```diff
@@ -362,8 +362,13 @@ const PayoutBankInformationForm = ({ isNew, getFieldName, host, fixedCurrency, i
   // Display spinner if loading
   if (loading) {
     return <StyledSpinner />;
-  } else if (!host.transferwise?.availableCurrencies && !fixedCurrency) {
-    return null;
+    // Make sure Wise is set at the host and also the available currencies list is valid
+  } else if (!host?.transferwise || (!host.transferwise?.availableCurrencies && !fixedCurrency)) {
```

Comment:
I don't get why `host?.transferwise` would be truthy and `host.transferwise?.availableCurrencies` falsy.

How can that happen?

Reply:
@znarf : I haven't seen personally a situation when `host?.transferwise` is true and `host.transferwise?.availableCurrencies` is false. I didn't want to remove/change the existing `!host.transferwise?.availableCurrencies && !fixedCurrency` part of the condition, as changing this condition before is probably why it started showing the warning in production. 🤔 

@kewitz : Can you please shed some insight on this as you added this condition before? 🤔  

But the existing condition `!host.transferwise?.availableCurrencies && !fixedCurrency` doesn't handle the case where `transferwise` is undefined; since then it can make the whole condition false and thus not showing the warning.

Reply:
@kewitz ?

Reply:
This condition was added 15 months ago to allow users to render the expense edit form even if they wanted to submit BANK_ACCOUNT expense after the host disconnected from Wise. Here's the PR https://github.com/opencollective/opencollective-frontend/pull/4343.

Can be simplified to:
```suggestion
  } else if (!host?.transferwise?.availableCurrencies && !fixedCurrency) {
```

Reply:
@kewitz : Thanks for the explanation; that helped a lot for me to understand the reasoning behind this one. 👍🏽

Reply:
Also on this note, I think we should return null on this condition; 

```
} else if (!host.transferwise?.availableCurrencies && !fixedCurrency) {
    return null;
  }
```
as we did before this PR, since otherwise in the expense form when the host disables Wise there's a possibility that the user sees the warning message, which is undesired. Rather they shouldn't see any warning. Am I correct on this @kewitz ? 🤔

Reply:
I'm not sure @SudharakaP, it depends on how that refactor to display the bank account payout form for every collective using our platform account as a fallback works.
It is my understatement that in the current state of our codebase, this message will only appear in an environment that has no Wise account connected, in practice this would be a broken dev environment.

Reply:
If we want to be on the safe side, we can add a condition to only display the message in dev environment. And console.log something otherwise.

Reply:
Good idea. I just added a condition for the expense flow case so that we don't show it on production accidentally. 

I also think we have to handle the two cases; 1) DetailsForm and 2) PayoutBankInformationForm separately as notice that in Details form the `host` object can be different (from querying the API) vs in the `PayoutBankInformationForm`; 

https://github.com/opencollective/opencollective-frontend/blob/5ad5688eb38fa38002de4a3d18b0bd183b2267af/components/expenses/PayoutBankInformationForm.js#L371

it can be either from the API or passed as an object; which we in `PayoutMethodForm`; thus I think we need to handle these two places separately rather than trying to handle everything in `PayoutBankInformationForm` which might work but prone to more problems. Let me know if you disagree with this and I am happy to change it. 🙂

Reply:
That doesn't make sense to me. We've been there multiple times, the `host` we're testing in `DetailsForm` is the same as `PayoutBankInformationForm`. We're displaying the message after `host = data?.host || host; `

Reply:
@znarf : So for example over here; 

https://github.com/opencollective/opencollective-frontend/blob/64b442b9fd233fb743442d9ddf1fa5ef5488cd26/components/edit-collective/sections/BankTransfer.js#L207

the `host` is an object `{ slug: TW_API_COLLECTIVE_SLUG }`. And we have the default value, `opencollective-host` for this env variable. So `data.host` can be `undefined`; because we only run the query;

https://github.com/opencollective/opencollective-frontend/blob/3cced4c629ea3cd3adaf4b6a65c13c190375ba0a/components/expenses/PayoutBankInformationForm.js#L363-L368

for `Boolean(fixedCurrency || host.transferwise?.availableCurrencies)`. For the default case in dev environment this is always true as the `fixedCurrency` is `USD` and the `host.transferwise` is undefined. If we were to handle this in one place we need to change this query to not skip on this condition. However I don't think that's good since `PayoutBankInformationForm` is used in multiple places and we don't want to run this every time. 🤔

Reply:
I'm sorry but I don't see how your comment is answering my feedback.

>  The host we're testing in DetailsForm is the same as the one in PayoutBankInformationForm.

We're effectively doing `host = data?.host || host; ` before anything else and we're using that same value in  DetailsForm and within PayoutBankInformationForm. I don't see why we're talking about anything else, like discussing if `data.host` is set or not.

Reply:
The `host` we are passing to the `DetailsForm` is the same one we calculated in; 

https://github.com/opencollective/opencollective-frontend/blob/3cced4c629ea3cd3adaf4b6a65c13c190375ba0a/components/expenses/PayoutBankInformationForm.js#L371

However after passing that value, based on the slug it's re-calculated by the query (within `DetailsForm.js`);

https://github.com/opencollective/opencollective-frontend/blob/3cced4c629ea3cd3adaf4b6a65c13c190375ba0a/components/expenses/PayoutBankInformationForm.js#L222-L225

So as an example in dev environment, the value calculated at `host = data?.host || host` can be; `{ slug: TW_API_COLLECTIVE_SLUG }` whereas the one used in the `DetailsForm` can by something like; `{ __typename: "Host", transferwise: null }`. Does that make sense? So essentially, the `host` value we're testing in the PayoutBankInformationForm and the DetailsForm can be different.

Reply:
Ok, that would have helped me if you would have said from the beginning "No, the `host` is not the same object, because in `DetailsForm` we're fetching `requiredFieldsQuery` and this is what we're looking at (`data.host`), not the `host` passed as prop."

Reply:
Sorry, yes I should have communicated better. 🤔